java/core/pom.xml

```diff
@@ -75,6 +75,10 @@
       <groupId>org.threeten</groupId>
       <artifactId>threeten-extra</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.tdunning</groupId>
```

Comment:
This reminds me the following Spark PR. In the community, it's a little hard to add this kind of dependency without a significant justification.
- https://github.com/apache/spark/pull/26197/files#diff-283ead138b37209525130359830c4c6eeafaf5c6639b0df82b770389d24002bfR134

Reply:
Do we have some benchmark result for this enhancement?
> For example it can assist in filtering conditions and finding faster paths for pruning.

Reply:
> This reminds me the following Spark PR. In the community, it's a little hard to add this kind of dependency without a significant justification.
> 
> * https://github.com/apache/spark/pull/26197/files#diff-283ead138b37209525130359830c4c6eeafaf5c6639b0df82b770389d24002bfR134

I understand the point and the fear of not being maintained. But the actual algorithm is short and easy to understand.

> https://github.com/addthis/stream-lib/blob/master/src/main/java/com/clearspring/analytics/stream/quantile/TDigest.java

Can I copy this file into the ORC project instead of the dependency?

Reply:
> Do we have some benchmark result for this enhancement?
> 
> > For example it can assist in filtering conditions and finding faster paths for pruning.

@dongjoon-hyun, Currently pr is still in draft form. I will add benchmarking, unit testing and documentation. It might be a bit slow because I need to finish the work at hand first. 😃

Reply:
I agree with @dongjoon-hyun that introducing a new dependency adds more complexity to this project. It looks like t-digest has been widely adopted by many projects and big name companies: https://www.sciencedirect.com/science/article/pii/S2665963820300403. It would be good if we can make it as a plugin instead of a direct dependency.

Reply:
@guiyanakuang I am also curious of your preference of t-digest over DataSketches which is also an Apache project shipped with similar feature.

Reply:
@wgtmac  I first learned about TDigest from this [issue](https://issues.apache.org/jira/projects/ORC/issues/ORC-98?filter=allopenissues), which inspired me to. So I don't actually have a preference, I see that DataSketches has implementations in different languages, and if it is well compatible with reading and writing serialisation in different languages, then I am in favour of using DataSketches. I need some time to learn it. :)